document-portal/document-portal.c

```diff
@@ -1416,6 +1416,103 @@ portal_list (GDBusMethodInvocation *invocation,
   return TRUE;
 }
 
+const char *
+get_host_path_internal (GDBusMethodInvocation *invocation,
+                        XdpAppInfo            *app_info,
+                        const char            *id,
+                        GError                **error)
+{
+  g_autoptr(PermissionDbEntry) entry = NULL;
+
+  XDP_AUTOLOCK (db);
+
+  entry = permission_db_lookup (db, id);
+
+  if (!entry)
+    {
+      if (error != NULL && *error == NULL)
+        {
+          g_set_error (error,
+                      XDG_DESKTOP_PORTAL_ERROR,
+                      XDG_DESKTOP_PORTAL_ERROR_INVALID_ARGUMENT,
+                      "Invalid ID passed (%s)", id);
+        }
+
+      return NULL;
+    }
+
+  if (!xdp_app_info_is_host (app_info))
+    {
+      g_autofree const char **apps = NULL;
+      const char *app_id = NULL;
+      gboolean app_found = FALSE;
+      int i;
+
+      app_id = xdp_app_info_get_id (app_info);
+
+      apps = permission_db_entry_list_apps (entry);
+      for (i = 0; apps[i] != NULL; i++)
```

Comment:
`for (size_t i = 0; apps[i] != NULL; i++)`

Reply:
Ok, changing this for the third time. I had `size_t i` as suggested by @ebassi, then @swick wanted the declaration of `i` be at the beginning with the rest of declarations and now you want me to change it again :).

Reply:
Yeah sorry for this back and forth. I've been preferring this syntax because it keeps `i` scoped to the loop.

Reply:
We're not declaring variables in loops in e.g. mutter and this project mostly doesn't either (there are a few exceptions). Do we want to start doing that? Should we get a coding style document somewhere?

Reply:
I think it's good if we start doing that. I don't think we have x-d-p coding style documented, but that would be good to have as well.